### DISCUSSION TITLE
Hyper based HTTP proxy ?

### DISCUSSION THREAD

**Discussion by tomershaniii** [hyperium/hyper#2747] [Q&A]
Hi,
Looking for an HTTP proxy implementation in rust, with support for HTTP/1.1 pipelining.
Was wondering if a hyper based implementation exists or at all makes sense (as hyper is primarily designed to be a server)

**Comment by seanmonstar** [comment]
hyper is both a server and a client. Many proxies have been written using it. They usually do something specific (for example, linkerd2). Starting a new proxy in hyper is quite simple: create a hyper server that passes the request on to a hyper client, and send that response back to the original request.

hyper does not support HTTP/1.1 pipelining, since it's a deprecated HTTP feature. It's better to use HTTP/2.

**Reply by tomershaniii** [reply]
Got it, thank you Sean, will give it a try
